File: new-backend/api/internal/models/politician.go
package models

// Politician is the DB document stored in MongoDB.
type Politician struct {
	ID           string  `bson:"_id"          json:"id"`
	Name         string  `bson:"name"         json:"name"`
	Gender       int     `bson:"gender"       json:"gender"`
	ImageUrl     *string `bson:"imageUrl"     json:"imageUrl"`
	PartyId      string  `bson:"partyId"      json:"-"`
	Active       bool    `bson:"active"       json:"active"`
	WorkLocation int     `bson:"workLocation" json:"workLocation"`
}

// PoliticianWithParty is returned by aggregation queries that join to the parties collection.
type PoliticianWithParty struct {
	ID           string  `bson:"_id"          json:"id"`
	Name         string  `bson:"name"         json:"name"`
	Gender       int     `bson:"gender"       json:"gender"`
	ImageUrl     *string `bson:"imageUrl"     json:"imageUrl"`
	PartyId      string  `bson:"partyId"      json:"-"`
	Active       bool    `bson:"active"       json:"active"`
	WorkLocation int     `bson:"workLocation" json:"workLocation"`
	Party        *Party  `bson:"party"        json:"party"`
}

File: Backend/api/app/models/politician.py
import uuid
from sqlalchemy import Column, String, Boolean, Integer, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship

from app.database import Base


class Politician(Base):
    __tablename__ = "Politicians"

    Id = Column("Id", UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    Name = Column("Name", String, nullable=False, default="")
    Gender = Column("Gender", Integer, nullable=False, default=0)
    ImageUrl = Column("ImageUrl", String, nullable=True)
    PartyId = Column("PartyId", UUID(as_uuid=True), ForeignKey("Parties.Id"), nullable=False)
    Active = Column("Active", Boolean, nullable=False, default=False)
    WorkLocation = Column("WorkLocation", Integer, nullable=False, default=0)

    party_rel = relationship("Party", back_populates="politicians", lazy="joined")
    votes = relationship("Vote", back_populates="politician_rel")
